torch/_inductor/codegen/cpp_wrapper_mps.py

```diff
@@ -133,48 +226,72 @@ def get_device_include_path(device: str) -> str:
-        }
+        AOTIMetalShaderLibraryHandle mps_lib_0_handle = nullptr;
+        AOTIMetalKernelFunctionHandle mps_lib_0_kernel_handle = nullptr;
+
         AOTIMetalKernelFunctionHandle get_mps_lib_0_handle() {
-            static const auto handle = AOTIMetalKernelFunctionHandle(get_mps_lib_0().get());
-            return handle;
+            if (mps_lib_0_kernel_handle == nullptr) {
+                if (mps_lib_0_handle == nullptr) {
+                    aoti_torch_mps_create_shader_library(R"MTL(...shader_source...)MTL", &mps_lib_0_handle);
+                }
+                aoti_torch_mps_get_kernel_function(mps_lib_0_handle, "generated_kernel", &mps_lib_0_kernel_handle);
+            }
+            return mps_lib_0_kernel_handle;
         }
         ```
         """
 
+        # Add shimified handles and functions
+        shader_libraries: OrderedSet[str] = OrderedSet()
         for line in self.lines:
             if not isinstance(line, KernelCallLine):
                 continue
             if line.device.type != "mps":
                 continue
 
-            # Only add handle definition once
+            # Extract library name from kernel name (e.g., "mps_lib_0" from kernel calls)
             if line.kernel_name not in self._used_kernel_names:
                 self._used_kernel_names.add(line.kernel_name)
+                shader_libraries.add(line.kernel_name)
 
-                self.prefix.writeline(
-                    f"const std::shared_ptr<at::native::mps::MetalKernelFunction> get_{line.kernel_name}() {{"
-                )
-                self.prefix.writeline(
-                    f'    static const auto func = {line.kernel_name}.getKernelFunction("generated_kernel");'
-                )
-                self.prefix.writeline("    return func;")
-                self.prefix.writeline("}")
+        # NOTE: For shimified version, we expect the shader source constant to be generated
+        # by the existing MPS shader generation process, but instead of instantiating the
+        # DynamicMetalShaderLibrary directly, we'll use our shim functions.
+        # The existing codegen should produce something like:
+        # const char* mps_lib_0_source = R"MTL(...shader_source...)MTL";
+        # instead of:
+        # at::native::mps::DynamicMetalShaderLibrary mps_lib_0(R"MTL(...shader_source...)MTL");
 
-                self.prefix.writeline(
-                    f"AOTIMetalKernelFunctionHandle get_{line.kernel_name}_handle() {{"
-                )
-                self.prefix.writeline(
-                    f"    static const auto handle = AOTIMetalKernelFunctionHandle(get_{line.kernel_name}().get());"
-                )
-                self.prefix.writeline("    return handle;")
-                self.prefix.writeline("}")
+        # Generate shimified declarations and functions for each library
+        for lib_name in shader_libraries:
+            # Add handle declarations
+            self.prefix.writeline(
```

Comment:
You can use `self.prefix.splice` to write multiple lines. The current way is hard to read.

Reply:
done!